### PR TITLE
Added file constraint check

### PR DESCRIPTION
**Pull request by webbyfox**

### CODE REVIEW

File: djangocms_versioning_filer/helpers.py [hunk 1]
```diff
@@ -74,3 +75,35 @@ def check_file_exists_in_folder(file_obj):
...
+def filename_exists(request, folder_id=None):
+    from filer.models import Folder, File
+
+    FILE_EXISTS = 'File name already exists'
+
+    try:
+        # Get folder
+        folder = Folder.objects.get(pk=folder_id)
+    except Folder.DoesNotExist:
+        return
```
[the remark this answers is not in the document]
Comment (monikasulik) [reply djangocms_versioning_filer/helpers.py]: The comment you've added just describes what's already visible :) What I'd want to know is not what it's doing but why it's doing that.

File: tests/test_filename_exists.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import json
+from .base import BaseFilerVersioningTestCase
+from django.urls import reverse
```
Comment (monikasulik) [inline tests/test_filename_exists.py]: The imports aren't pep8 compliant :8ball:

File: tests/test_filename_exists.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        with self.login_user_context(self.superuser):
+            response = self.client.post(
+                reverse('admin:filer-check_file_constraints',
+                        kwargs={'folder_id': self.folder.id})
+                + '?filename={}'.format('test.txt'),
+
+            )
+            self.assertIn('success', json.loads(response.content))
+            self.assertEqual(False, json.loads(response.content)['success'])
+            self.assertIn('message', json.loads(response.content))
```
Comment (monikasulik) [inline tests/test_filename_exists.py]: Shouldn't you be checking the content of the message as well?

File: tests/test_filename_exists.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    def test_filename_not_exists(self):
+        with self.login_user_context(self.superuser):
+            response = self.client.post(
+                reverse('admin:filer-check_file_constraints',
+                        kwargs={'folder_id': self.folder.id})
+                + '?filename={}'.format('test.txt'),
+
+            )
+            self.assertIn('success', json.loads(response.content))
+            self.assertEqual(True, json.loads(response.content)['success'])
```
Comment (monikasulik) [inline tests/test_filename_exists.py]: I think there should also be: `self.assertNotIn('message', json.loads(response.content))`

File: djangocms_versioning_filer/helpers.py [hunk 5]
```diff
@@ -74,3 +75,31 @@ def check_file_exists_in_folder(file_obj):
     return check_file_label_exists_in_folder(
         file_obj.label, file_obj.folder, exclude_file_pks=exclude_file_pks,
     )
+
+
+def filename_exists(request, folder_id=None):
+    from filer.models import Folder, File
+
+    FILE_EXISTS = 'File name already exists'
```
Comment (chaosk) [inline djangocms_versioning_filer/helpers.py]: `_()`

**Pull request closed without merging**